src/utils/optionsparser.cc

```diff
@@ -460,6 +463,19 @@ void IntOption::SetVal(OptionsDict* dict, const ValueType& val) const {
   dict->Set<ValueType>(GetId(), val);
 }
 
+void IntOption::ValidateIntString(const std::string& val) {
+  const int base = 10;
+  char *end;
+  errno = 0;
+  std::strtol(val.c_str(), &end, base);
```

Comment:
This function already converts the value, why not just return it? :)
Also the more "modern" way of the same function is https://en.cppreference.com/w/cpp/utility/from_chars, consider using it instead.
```cpp
uint64_t result;
const auto end = val.data() + val.size();
auto [ptr, err] = std::from_chars(val.data(), end, &result);
if (err != std::errc() || ptr != end) {
	throw('Blah blah');
}
return result;
```